unix/variants/coverage: Add test for manifest freeze_mpy().

jimmo opened the pull request:
Includes the fix from #9315 (@dmazzella) and adds a test for this case.

This adds the frozentest.mpy (currently used by ports/minimal) to the unix coverage build and tests it in the coverage_extra test. Moved frozentest.mpy into a common location now that it's shared with three things (minimal, powerpc, unix-coverage).

Also fixes two bugs that these new tests picked up:
 - File extension matching in manifestfile.py.
 - Handling of freeze_mpy results in makemanifest.

_This work was funded through GitHub Sponsors._

dpgeorge commented:
TODO: consolidate `ports/powerpc/frozentest.mpy` and `ports/minimal/frozentest.mpy` to `tests/` and use that in this PR.

jimmo committed:
tests/frozen: Move frozentest.mpy from ports/ to tests/.

frozentest.mpy was previously duplicated in ports/minimal and
ports/powerpc.

This needs to be re-generated on every .mpy version increase, so might as
well just have a single copy of it.

Signed-off-by: Jim Mussared <jim.mussared@gmail.com>

jimmo committed:
unix/variants/coverage: Add test for manifest freeze_mpy().

This uses the frozentest.mpy that is also used by ports/minimal.

Also fixes two bugs that these new tests picked up:
 - File extension matching in manifestfile.py.
 - Handling of freeze_mpy results in makemanifest.

This work was funded through GitHub Sponsors.

Signed-off-by: Jim Mussared <jim.mussared@gmail.com>

jimmo commented:
> TODO: consolidate `ports/powerpc/frozentest.mpy` and `ports/minimal/frozentest.mpy` to `tests/` and use that in this PR.

Done. Updated branch and the PR description.

dpgeorge commented:
Thanks, merged in 6ecdf1a240e43bd60c824c7efd575c0a82d02d7e and 920da9c5e30da58d5f77b2c5e28c74e3670e6a3b
